.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,29 @@
+name: 'CodeQL'
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏since we are now using GitHub action, we might want to setup dependabot to check for actions updates ([example](https://github.com/BenoitZugmeyer/eslint-plugin-html/blob/main/.github/dependabot.yml#L8-L12))

Reply:
👍, let's do that in a following PR